### PR TITLE
Fix document for default mkldnn backend

### PR DESCRIPTION
**Pull request by TaoLv**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: MKLDNN_README.md [hunk 1]
```diff
@@ -39,10 +39,14 @@ cd incubator-mxnet
...
 
 ```
-make -j $(nproc) USE_OPENCV=1 USE_MKLDNN=1 USE_BLAS=mkl USE_INTEL_PATH=/opt/intel
+make -j $(nproc) USE_OPENCV=1 USE_BLAS=mkl USE_INTEL_PATH=/opt/intel
 ```
 
-If you don't have the full [MKL](https://software.intel.com/en-us/intel-mkl) library installation, you might use OpenBLAS as the blas library, by setting USE_BLAS=openblas.
+For x86 Linux, it will build MXNet with MKL-DNN backend by default. If you want to disable MKL-DNN backend, you can set `USE_MKLDNN=0` explicitily in the command line. Or you can `export MXNET_MKLDNN_ENABLED=0` before running your model if the MXNet binary is already built with MKL-DNN.
+
+For other platforms, you can set `USE_MKLDNN=1` explicitily in the command line to enable MKL-DNN backend. Otherwise, MKL-DNN backend will not be built. Please note that, along with the development and maturing of MKL-DNN backend, the default setting for `USE_MKLDNN` may be changed and MKL-DNN backend will become the default backend for other platforms too.
```
Comment (ciyongch) [inline MKLDNN_README.md]: For x86 Linux, MXNet will be built with MKL-DNN backend by default.  .....
....
For other platforms, MXNet will be built WITHOUT MKL-DNN backend by default. You can set `USE_MKLDNN=1` explicitly in the command line to enable MKL-DNN backend. Please note that....

explicitily -> explicitly

**Review by ciyongch — COMMENTED** [review]
explicitily -> explicitly, a minor typo :)

File: MKLDNN_README.md (hunk 1 above)
Comment (TaoLv) [reply MKLDNN_README.md]: Thank you. Fixed.

**Review by ciyongch — APPROVED** [review]

**Comment by TaoLv** [comment]
@xinyu-intel @pengzhao-intel Do we still need to have these document changes on master branch.? Anything else need to be added?
